lib/CL/devices/basic/basic.c

```diff
@@ -719,14 +719,21 @@ pocl_basic_fill_rect (void *data,
                 + buffer_slice_pitch * k, fill_pixel, pixel_size);
 }
 
+typedef unsigned long v2ul __attribute__ ((vector_size (16)));
```

Comment:
1) If you want i can revert to nested for loops.

2) AFAICT we check the alignment, we allocate buffers aligned, and  clEnqueueFillBuffer() requires both size & offset to be multiple of sizeof(pattern).

3) I did implement the original switch/case, what do you find counterproductive about it ?

4) no i did not measure anything, the code with vectors is simpler / shorter. I saw  no reason to not use vectors.